### PR TITLE
Add CFcompliance and data_validation to the process options

### LINKED ISSUE #159: xclim.core.utils.ValidationError: time series is not recognized as daily
## Description

During execution of process `BCCAQv2_heat_wave_frequency_gridpoint`, we obtain this traceback:

```
Traceback (most recent call last):
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/pywps/app/Process.py", line 248, in _run_process
    self.handler(wps_request, wps_response)  # the user must update the wps_response.
  File "/code/finch/processes/wps_base.py", line 37, in _handler_wrapper
    return self.wrapped_handler(request, response)
  File "/code/finch/processes/wps_bccaqv2_heatwave.py", line 155, in _handler
    out = compute_indices(self, self.indices_process.xci, inputs)
  File "/code/finch/processes/utils.py", line 163, in compute_indices
    out = func(**kwds)
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/indicator.py", line 456, in __call__
    self._bind_call(self.datacheck, **das)
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/indicator.py", line 538, in _bind_call
    return func(*ba.args, **ba.kwargs)
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/indicator.py", line 820, in datacheck
    datachecks.check_daily(da)
  File "<boltons.funcutils.FunctionBuilder-115>", line 2, in check_daily
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/options.py", line 98, in run_check
    return _run_check(func, DATA_VALIDATION, *args, **kwargs)
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/options.py", line 90, in _run_check
    raise err
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/options.py", line 83, in _run_check
    func(*args, **kwargs)
  File "/opt/conda/envs/finch/lib/python3.9/site-packages/xclim/core/datachecks.py", line 45, in check_daily
    raise ValidationError("time series is not recognized as daily.")
xclim.core.utils.ValidationError: time series is not recognized as daily.
```

## Environment

* Finch version used, if any: 0.7.0

## Steps to Reproduce

Make a request to finch using these params on bccaqv2 data:

```
"inputs": [
		{
			"id": "lat",
			"data": "45.5837167538"
		},
		{
			"id": "lon",
			"data": "-73.7505505823"
		},
		{
			"id": "thresh_tasmin",
			"data": "16 degC"
		},
		{
			"id": "thresh_tasmax",
			"data": "22 degC"
		},
		{
			"id": "freq",
			"data": "YS"
		},
		{
			"id": "start_date",
			"data":"2000"
		},
		{
			"id":"end_date",
			"data":"2002"
		},
		{
			"id": "output_format",
			"data": "netcdf"
		}
	]
```

### PR DESCRIPTION
**Pull request by aulemahal**
## Overview

This PR fixes #159

Changes:

* Added "cf_compliance" and "data_validation" to the common inputs for xclim indicator processes. 
* Decreased the strictness level for "data_validation" in "BCCAQV2HeatWave" (from "raise" to "warn").

## Related Issue / Discussion
Opening the process with birdy (wpsclient) I got a weird bug where the function signature is super strange:
```
Signature:
wps.bccaqv2_heat_wave_frequency_gridpoint(
    missing_options,
    lon,
    lat=None,
    check_missing='any',
    cf_compliance='warn',
    data_validation='warn',
    thresh_tasmin='22.0 degC',
    thresh_tasmax='30 degC',
    window=3,
    freq='YS',
    variable=None,
    y0=None,
    y1=None,
    output_format='netcdf',
    output_formats=None,
)
```
How come "missing_options" is now first and positional??? I haven't touched it!

### CODE REVIEW

**Review by huard — APPROVED** [review]

**Comment by aulemahal** [comment]
@huard my last commit fixes the signature problem when I use birdy. For me it also happens on master. 
Can you confirm if this was really a bug? I can revert the commit if not.

May @MatProv wants to test against this branch, before we merge?

**Comment by aulemahal** [comment]
After discussion, it seems that the signature problem was indeed a bug that we had simply not seen before. I will merge when the checks complete.

**Comment by tlogan2000** [comment]
FYI the analysis page does more than heat wave calculations these days.  I think there may be other processes to edit as well..?.

**Comment by tlogan2000** [comment]
@MatProv can you identify which processes are called by the anaysis page?

**Comment by aulemahal** [comment]
In that case, if the called processes are not BCCAQv2-dependent I would suggest passing  `data_validation='warn'` in the calls themselves. Raising an error on non-daily data is something we want in most cases!

**Comment by MatProv** [comment]
@tlogan2000 @aulemahal 
The analysis page use these processes:
```
wps_bccaqv2_heatwave
wps_ensemble_indices_bbox
wps_ensemble_indices_point
```

And potentially these ones (not sure, might depend on subsetting logic):
```
wps_xsubset_bbox_dataset
wps_xsubset_bbox
wps_xsubset_point_dataset
wps_xsubset_point
```

File: docs/source/notebooks/finch-usage.ipynb
Comment (tlvu) [inline docs/source/notebooks/finch-usage.ipynb:79]: FYI, this breaks Jenkins, will need a new Finch deployed to production

**Pull request merged**
